src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
-    List<Object> get();
+    List<? super T> get();
```

Comment:
The `get()` method should return `List<T>` instead of `List<? super T>`. This ensures that the method returns a list of the specific type `T` or its subtypes, which aligns with the task requirements for producing specific machine models.